### DISCUSSION TITLE
machine.I2C issue with Micropython 1.17

### DISCUSSION THREAD

**Discussion by clharting** [BIPES/BIPES#75] [Q&A] [answered]
Thank you for creating this IDE! I'm still figuring out a lot of the features, but it looks great and it's helping my quickly code for a number of devices in Micropython.
I have encountered an issue with machine.I2C.
I am using a Wemos D1 Mini (ESP8266) with an OLED display (SSD1306). When I try to init the display using the "Init I2C SSD1306 OLED Display" block, I get:
=== from machine import Pin
=== from machine import I2C
=== i2c=I2C(1, scl=Pin(14), sda=Pin(12))
Traceback (most recent call last):
  File "<stdin>", line 7, in <module>
TypeError: extra keyword arguments given
It doesn't seem to matter what value I put in for the I2C number.
I believe there was a change in the Micropython code, such that the I2C ID is no longer used. But, I can't leave the ID field blank, as it defaults to 0.
If I type the following in the console, I don't get an error:
>>> from machine import I2C, Pin
>>> i2c=I2C(Pin(14),Pin(12))
Can you fix this, please? Or, am I missing something?
Thanks!

Code:
from machine import Pin
from machine import I2C
import ssd1306
from time import sleep

i2c=I2C(1, scl=Pin(14), sda=Pin(12))
oled_width = 128
oled_height = 64
oled = ssd1306.SSD1306_I2C(oled_width, oled_height, i2c)
oled.fill(0)
oled.text('Hello', 10, 40)
oled.show()

#Code automatically generated by BIPES (http://www.bipes.net.br)

**Comment by gastmaier** [comment, marked as answer]
Hi!
can you please provide the micropython version running in your board with:
```
import os
os.uname()
```
Because when I run the suggested code with the latest release (v1.17.0), I get:
```
Warning: I2C(-1, ...) is deprecated, use SoftI2C(...) instead
```
As described in the current [machine.I2C documentation](https://docs.micropython.org/en/latest/library/machine.I2C.html), the I2C should provide an ID. Without IDs, there is Software I2C, which is constructed like this:
```
from machine import Pin, SoftI2C
i2c=SoftI2C( scl=Pin(14), sda=Pin(12))
```
I can change the block if needed to use SoftI2C instead (or make it selectable), just confirm me if SoftI2C works for you and the micropython version you are running :smile: .

**Reply by clharting** [reply]
>>> os.uname()
(sysname='esp8266', nodename='esp8266', release='2.0.0(5a875ba)', version='v1.17 on 2021-09-02', machine

**Reply by clharting** [reply]
Ah, using -1 as the I2C ID works!

**Reply by gastmaier** [reply]
Good. I will include a note/description to the block about this behaviour later.

Thanks for the info!

**Reply by Fa5g** [reply]
I´m experiencing the same issue, with (sysname='esp8266', nodename='esp8266', release='2.2.0-dev(9422289)', version='v1.21.0 on 2023-10-05', machine='ESP module with ESP8266'), I tried with id argument -1, always gives "TypeError: extra keyword arguments given" and without id argument tested in console, it works.

**Reply by marcusdemian** [reply]
Need help on this issue. I am trying to use -1 as I3C ID on ESP8266 NodeMCU and it is not working.
Micropython version is 1.23
As far I understood from other threads, Nodemcu I2C has only 2 params as input.
How can this be solved ?

**Comment by florisniehaus** [comment]
I had similar problems. If I download Micropython V1.13 or prior versions, the I2C works!. 
Micropyrhon V1.17 or later does not work. 
Is it possible that someone can update the I2C SSD1307 OLED Blocks to work with the newest Micropython versions. 
Can somebody help, of give directions how to change it. Thanks

**Comment by jetannenbaum** [comment]
I'm running MicroPython v1.22.1 on 2024-01-05; Raspberry Pi Pico with RP2040.  Just tested with I2C and the code works fine with setting the ID.

This is the generated code:
![BipesCode](https://github.com/BIPES/BIPES/assets/865106/b133c293-be64-4dbb-950e-1cf88994da74)
